### PR TITLE
filter school

### PR DESCRIPTION
**Pull request by Padillahirpan**
filter school

### CODE REVIEW

File: app/src/main/java/com/maro/sesamamuslim/views/school/SchoolFragment.kt [hunk 1]
```diff
@@ -146,7 +149,24 @@ class SchoolFragment : AbstractSchoolFragment(), FragmentNavigation, SchoolAdapt
...
+        val dialog = CityFilterDialogFragment.newInstance(this)
+        activity?.let {
+            dialog.show(it.supportFragmentManager, "bottom-sheet-city-filter")
+        }
+    }
+
+    override fun onItemCityClicked() {
+        val dialog = SchoolFilterDialogFragment.newInstance(this)
+        activity?.let {
+            dialog.show(it.supportFragmentManager, "bottom-sheet-filter")
```
Comment (wanharaderta) [inline app/src/main/java/com/maro/sesamamuslim/views/school/SchoolFragment.kt:169]: standar codingan jangan harcode message atau naming, coba pindahin ke class SchoolFilterDialogFragment

File: app/src/main/java/com/maro/sesamamuslim/views/school/dialog/CityFilterDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.maro.sesamamuslim.R
+import com.maro.sesamamuslim.views.school.adapter.CityAdapter
+import kotlinx.android.synthetic.main.layout_filter_city_bottom_sheet.*
+
+class CityFilterDialogFragment : AbstractSchoolFilterDialogFragment(), CityAdapter.CityListener {
+
+    private val cityAdapter by lazy { CityAdapter(mutableListOf(), onClick = this) }
+    private lateinit var data: FilterListener
+
+    companion object {
```
Comment (wanharaderta) [inline app/src/main/java/com/maro/sesamamuslim/views/school/dialog/CityFilterDialogFragment.kt:17]: move paling bawah

File: app/src/main/res/layout/item_city.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <TextView
```
Comment (wanharaderta) [inline app/src/main/res/layout/item_city.xml:7]: move androidx

File: school/src/main/java/com/maro/school/view/AbstractSchoolFragment.kt [hunk 4]
```diff
@@ -22,12 +23,14 @@ abstract class AbstractSchoolFragment : BaseFragment<SchoolViewModel>() {
...
 
 
-    override val viewModel: SchoolViewModel by viewModel()
+    override val viewModel: SchoolViewModel by sharedViewModel()
 
     override fun viewInitialization(view: View) {
         super.viewInitialization(view)
 
-        viewModel.getSchoolsPage(page, keywordSearch)
+        if (!viewModel.isFilter) viewModel.getSchoolsPage(page, keywordSearch)
```
Comment (wanharaderta) [inline school/src/main/java/com/maro/school/view/AbstractSchoolFragment.kt:31]: !viewModel.isFilter ganti jadi viewModel.isFilter.not()

File: app/src/main/java/com/maro/sesamamuslim/views/school/SchoolFragment.kt (hunk 1 above)
Comment (Padillahirpan) [reply app/src/main/java/com/maro/sesamamuslim/views/school/SchoolFragment.kt:169]: oiyah class name yah

File: school/src/main/java/com/maro/school/view/AbstractSchoolFragment.kt (hunk 4 above)
Comment (Padillahirpan) [reply school/src/main/java/com/maro/school/view/AbstractSchoolFragment.kt:31]: okay

File: app/src/main/res/layout/item_city.xml (hunk 3 above)
Comment (Padillahirpan) [reply app/src/main/res/layout/item_city.xml:7]: okay

File: app/src/main/java/com/maro/sesamamuslim/views/school/dialog/CityFilterDialogFragment.kt (hunk 2 above)
Comment (Padillahirpan) [reply app/src/main/java/com/maro/sesamamuslim/views/school/dialog/CityFilterDialogFragment.kt:17]: companion object nya doang yah ?

Comment (wanharaderta) [reply app/src/main/java/com/maro/sesamamuslim/views/school/dialog/CityFilterDialogFragment.kt:17]: yoi

**Pull request merged**